hera_cal/abscal.py

```diff
@@ -129,6 +129,137 @@ def abs_amp_logcal(model, data, wgts=None, verbose=True, return_gains=False, gai
         return {ant: np.exp(fit['eta_{}'.format(ant[1])]).astype(np.complex) for ant in gain_ants}
 
 
+def abs_amp_lincal(model, data, wgts=None, verbose=True, return_gains=False, gain_ants=[],
+                   conv_crit=None, maxiter=100):
+    """
+    calculate absolute (array-wide) gain amplitude scalar
+    with a linear (or linearized) solver using the equation:
+
+    V_ij,xy^data = A_x A_y * V_ij,xy^model
+
+    where {i,j} index antenna numbers and {x,y} index polarizations
+    of the i-th and j-th antennas respectively. When no cross-polarized
+    visibilities are involved, A^2 is solved for linearly for both real
+    and imaginary parts simultaneously as separate equations. Otherwise,
+    we have to use a linear-product solving algorithm, using abs_amp_logcal
+    as a starting point.
+
+    Parameters:
+    -----------
+    model : visibility data of refence model, type=DataContainer
+            keys are antenna-pair + polarization tuples, Ex. (1, 2, 'nn').
+            values are complex ndarray visibilities.
+            these must be 2D arrays, with [0] axis indexing time
+            and [1] axis indexing frequency.
+
+    data : visibility data of measurements, type=DataContainer
+           keys are antenna pair + pol tuples (must match model), values are
+           complex ndarray visibilities matching shape of model
+
+    wgts : weights of data, type=DataContainer, [default=None]
+           keys are antenna pair + pol tuples (must match model), values are real floats
+           matching shape of model and data
+
+    return_gains : boolean. If True, convert result into a dictionary of gain waterfalls.
+
+    gain_ants : list of ant-pol tuples for return_gains dictionary
+    
+    conv_crit : A convergence criterion below which to stop iterating LinProductSolver. 
+                Converegence is measured L2-norm of the change in the solution of the 
+                variables divided by the L2-norm of the solution itself.
+                Default: None (resolves to machine precision for inferred dtype).
+                N.B. Only used when data and model include cross-polarized visibilities.
```

Comment:
had to look up what N.B. meant--I think just `Note:` is a little more commonly used in our codebase

Reply:
ok, changed it